### PR TITLE
Create minimal partial data layer scaffold starting off with Data Catalog for VEDA2 Refactor

### PR DESCRIPTION
**Pull request by sandrahoang686**
**Related Ticket:** 
* [[VEDA2 Refactor] Exploration - Data layer scaffolding ](https://github.com/NASA-IMPACT/veda-ui/issues/802)
* [[VEDA2 Refactor] - Data Catalog Core Feature Breakout](https://github.com/NASA-IMPACT/veda-ui/issues/892)
* 
### Description of Changes
This is a very minimal partial implementation of the data layer scaffolding to start the conversation and work on breaking out the feature components which will involve moving data logic out of the components into the data layer. This PR moves the datasets dependency out from the Data Catalog component to help kick off breaking out this Feature Component to be data agnostic

This Data Layer scaffold will be a WIP throughout the refactor and scaffold WILL EVOLVE. This is just to define a current place where data logic can be migrated too to start core feature breakouts

### Notes & Questions About Changes
@hanbyul-here thoughts?

### Validation / Testing
* Data Catalog Page should work the same

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
       "$components/*": ["app/scripts/components/*"],
       "$styles/*": ["app/scripts/styles/*"],
       "$utils/*": ["app/scripts/utils/*"],
-      "$context/*": ["app/scripts/context/*"]
+      "$context/*": ["app/scripts/context/*"],
+      "$data-layer/*": ["data-layer/*"]
```
Comment (oliverroick) [inline tsconfig.json:12]: To keep the repo clean, could we move this into from the repository root to a subdirectory like src?

```suggestion
      "$data-layer/*": ["src/data-layer/*"]
```

File: data-layer/datasets.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { datasets } from 'veda';
```
Comment (oliverroick) [inline data-layer/datasets.tsx:1]: Does this import the datasets as configured the instance config?

**Review by oliverroick — COMMENTED** [review]
I looked over this but I don’t fully understand these changes and how it will evolve in the future.

I left two questions in the code but my main question is about the `data-layer/datasets` module: How will this be used in the future? I’m assuming there will be more complexity in the module at some point; what functionality are we planning to implement in the module in the future?

**Comment by oliverroick** [comment]
Thanks @sandrahoang686 that's clear now. As for the work you've put in the PR, that looks all good to me. 

What we need to consider though is how the changes you plan in future relate to and affect to the[ discussion/spike for the parcel builder](https://github.com/NASA-IMPACT/veda-ui/issues/896) the @hanbyul-here is working on. From what Daniel is saying in the ticket, the parcel builder not only parses the content but also generates option lists that allow filtering for content. You mentioned data fetching in hooks, were you planning that this happens on-the-fly or are we still going to use a fair amount of pre-processing?

**Comment by sandrahoang686** [comment]
@oliverroick yeah I didn't realize how much the parcel resolver was doing... and I haven't thought about it. I'm thinking though and if I understood your question, seems like the parcel resolver is generating the list of taxonomies and datasets during build time but i'm thinking can we do something similar with something like NextJs's pre-rendering and static generation? So both? We initially fetch at build time with something like `getStaticProps` if using nextjs and this will not be managed by the hooks but the app can be pre-rendered and then fetch again on the client side when we need to?

What do you think?

File: tsconfig.json (hunk 1 above)
Comment (hanbyul-here) [reply tsconfig.json:12]: ☝️ Can we put `data-layer` in `app/scripts`? That is where all the 'scripts' are right now.

Comment (sandrahoang686) [reply tsconfig.json:12]: @oliverroick @hanbyul-here should I move it under `app/scripts/*` or create a new directory to do `src/data-layer/*`? I will say I personally dont understand why we have a dir under app named `/scripts` and why it was named that but maybe we can revisit naming later.. lmk which you both prefer here

Comment (oliverroick) [reply tsconfig.json:12]: If we want to keep it all in the same place it should go to `app/scripts`. But I agree, we should revisit that at some point, this `app/scripts` convention we're using in some projects is odd.

Comment (sandrahoang686) [reply tsconfig.json]: Updated, can I get a re-review?

**Review by oliverroick — APPROVED** [review]

**Pull request merged**